ProcessDoctor/ProcessProviderFactory.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.IO.Abstractions;
+using System.Runtime.InteropServices;
+using JetBrains.Diagnostics;
+using JetBrains.Lifetimes;
+using ProcessDoctor.Backend.Core.Interfaces;
+using ProcessDoctor.Backend.Linux.Proc;
+using ProcessDoctor.Backend.Windows.WMI;
+
+namespace ProcessDoctor;
+
+public sealed class ProcessProviderFactory
+{
+    public IProcessProvider Create(Lifetime lifetime)
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Prefer to use `OperatingSystem.IsWindows/IsLinux/IsMacOS` APIs, as they are more trimming friendly, and can help eliminating huge chanks of platform specific code, when compiled only for single platform.

Reply:
Thanks!
Fixed it